### PR TITLE
Cleanup some duplicated placeholder py:module docs

### LINKED ISSUE #111256: UNSTABLE pull / linux-docs / build-docs-python-false
Broken by https://github.com/pytorch/pytorch/pull/111108, cannot revert normally. Asking the author to forward fix.

cc @seemethere @malfet @pytorch/pytorch-dev-infra

### LINKED ISSUE #123068: UNSTABLE pull / linux-jammy-py3.8-gcc11 / test (docs_test)
This looks pretty unstable in trunk atm.  Marking it as unstable to look into this next week.

cc @ezyang @gchanan @zou3519 @kadeng @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @seemethere @malfet @pytorch/pytorch-dev-infra

### PR DESCRIPTION
**Pull request by huydhn**
Fixes https://github.com/pytorch/pytorch/issues/123068
Fixes https://github.com/pytorch/pytorch/issues/111256

While investigating the flaky doc build failure .w.r.t duplicated `torch.ao.quantization.quantize` docstring warning, i.e. https://github.com/pytorch/pytorch/actions/runs/8532187126/job/23376591356#step:10:1260, I discover an old but still open bug in Sphinx https://github.com/sphinx-doc/sphinx/issues/4459.  These warnings have always been there, but they are hidden because we are using `-j auto` to build docs with multiple threads.  It's just by chance that they start to surface now.

The issue can be reproduced by removing `-j auto` from https://github.com/pytorch/pytorch/blob/main/docs/Makefile#L5 and run `make html` locally.  Then, these warnings shows up consistently.  As `make html` treats warnings as errors, they will fail the build.

```
...
/data/users/huydo/conda/py3.8/lib/python3.8/site-packages/torch/ao/quantization/quantize.py:docstring of torch.ao.quantization.quantize.quantize:1: WARNING: duplicate object description of torch.ao.quantization.quantize, other instance in quantization, use :noindex: for one of them
/data/users/huydo/conda/py3.8/lib/python3.8/site-packages/torch/nn/parallel/data_parallel.py:docstring of torch.nn.parallel.data_parallel.data_parallel:1: WARNING: duplicate object description of torch.nn.parallel.data_parallel, other instance in nn, use :noindex: for one of them
/data/users/huydo/conda/py3.8/lib/python3.8/site-packages/torch/nn/utils/spectral_norm.py:docstring of torch.nn.utils.spectral_norm.spectral_norm:1: WARNING: duplicate object description of torch.nn.utils.spectral_norm, other instance in nn, use :noindex: for one of them
/data/users/huydo/conda/py3.8/lib/python3.8/site-packages/torch/nn/utils/weight_norm.py:docstring of torch.nn.utils.weight_norm.weight_norm:1: WARNING: duplicate object description of torch.nn.utils.weight_norm, other instance in nn, use :noindex: for one of them
/data/users/huydo/github/pytorch/docs/source/nn.rst:579: WARNING: duplicate object description of torch.nn.parallel.data_parallel, other instance in generated/torch.nn.functional.torch.nn.parallel.data_parallel, use :noindex: for one of them
/data/users/huydo/github/pytorch/docs/source/nn.rst:594: WARNING: duplicate object description of torch.nn.utils.spectral_norm, other instance in generated/torch.nn.utils.spectral_norm, use :noindex: for one of them
/data/users/huydo/github/pytorch/docs/source/nn.rst:595: WARNING: duplicate object description of torch.nn.utils.weight_norm, other instance in generated/torch.nn.utils.weight_norm, use :noindex: for one of them
/data/users/huydo/github/pytorch/docs/source/quantization.rst:1348: WARNING: duplicate object description of torch.ao.quantization.quantize, other instance in generated/torch.ao.quantization.quantize, use :noindex: for one of them
...
```

The fix is just to clean up those duplicated placeholder py:module docs, which were there because these modules didn't have any docs originally.

### CODE REVIEW

**Review by andrewor14 — APPROVED** [review]
Thanks for fixing this!

**Comment by huydhn** [comment]
Darn, now `make html` works, but `make coverage` complains a bit about missing entries, let me dig into this a bit more.

**Comment by clee2000** [comment]
I think the problem is that those files have a function that is exactly the same as the file name, so when the docs get generated, it generates a doc for the module/file quantize, and then it also tries to generate a doc with the same name for the function quantize, which is defined in quantize.py and is included in the `__all__` of the `__init__.py` file that is on the same level as quantize.py

I tested this locally by changing the spectral_norm function in the spectral_norm.py file to spectral_norm1 and the warning went away for spectral_norm

I haven't figure out how to resolve this other than renaming things

This also makes me wonder: when someone types `torch.ao.quantization.quantize`, are they expecting to call the function or are they expecting to import the module?  I guess syntax makes this obvious to the python interpreter?

**Review by malfet — APPROVED** [review]

File: .ci/pytorch/python_doc_push_script.sh [hunk 1]
```diff
@@ -105,7 +105,12 @@ if [ "$is_main_doc" = true ]; then
     echo undocumented objects found:
     cat build/coverage/python.txt
     echo "Make sure you've updated relevant .rsts in docs/source!"
-    exit 1
+    # NB: Due to some duplication of the following modules/functions, we keep
+    # them as expected failures for the time being instead of return 1
+    # - torch.nn.utils.weight_norm
+    # - torch.nn.utils.spectral_norm
+    # - torch.nn.parallel.data_parallel
+    # - torch.ao.quantization.quantize
```
Comment (clee2000) [inline .ci/pytorch/python_doc_push_script.sh:113]: I think you should change docs/conf.py to ignore these 4, it looks like this part of the coverage check is implemented ourself

Comment (huydhn) [reply .ci/pytorch/python_doc_push_script.sh:113]: I tried, but none of the `coverage_ignore_` works AFAICT https://www.sphinx-doc.org/en/master/usage/extensions/coverage.html like we need.  `make coverage` finishes successfully, but it always prints these out as a warning and the script treats that as extra failures.  This is not ideal, but is a low risk change IMO to keep trunk sane for now.  I plan to land this, then try to upgrade sphinx to a newer version to see if it helps.

Comment (clee2000) [reply .ci/pytorch/python_doc_push_script.sh:113]: https://github.com/pytorch/pytorch/blob/18c9d460682293bae23386e9bc40932eee1d8add/docs/source/conf.py#L3596-L3619

Comment (huydhn) [reply .ci/pytorch/python_doc_push_script.sh:113]: Ohhh, got it.  TIL, the script is the one responsible for that.  So, let me push a follow-up fix.

**Pull request closed without merging**